src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -191,6 +202,16 @@ private static void AddCollectionRuleTriggerSchema<TOptions>(GenerationContext c
 
             settingsProperty.Reference = context.AddTypeIfNotExist<TOptions>();
 
+            foreach (var propName in typeof(CollectionRuleDefaultsOptions).GetProperties().Select(x => x.Name))
+            {
+                settingsProperty.Reference.RequiredProperties.Remove(propName);
+            }
+
+            if (settingsProperty.Reference.RequiredProperties.Count > 0)
```

Comment:
This is almost correct; you have to check that the required properties don't have default values and there is no corresponding default property.

Reply:
I'm not sure I'm interpreting this correctly, so just to clarify what you mean:
1) If there exists a `[Required]` property that doesn't have a default value (as in `[DefaultValue(40)]`) and isn't one of the `CollectionRuleDefaults` properties, then mark `Settings` as `Required` - my confusion here is that the preceding loop removes any `[Required]` property that is also in `CollectionRuleDefaults`, and I'm not sure why a `[Required]` property would have a `[DefaultValue]` on it; or
2) If there exists a property that's `[Required]` but the `CollectionRuleDefaults` property for it hasn't been set, `Settings` is considered `Required`, but then is no longer considered `Required` if that `CollectionRuleDefaults` property is set - I'm not actually sure if this is feasible, but would be a more nuanced way of only showing settings as `Required` when it truly should be.

Hopefully that makes sense - I realized that `Required` is being used in two different contexts between the annotation and the schema, but I tried to use brackets to help with that.